mavros_extras/src/plugins/esc_telemetry.cpp

```diff
@@ -37,9 +95,22 @@ class ESCTelemetryPlugin : public plugin::PluginBase
 	{
 		PluginBase::initialize(uas_);
 
-		esc_telemetry_pub = nh.advertise<mavros_msgs::ESCTelemetry>("esc_telemetry", 10);
+		esc_telemetry_pub = nh.advertise<mavros_msgs::ESCTelemetry>("esc_telemetry", 1);
 
 		enable_connection_cb();
+
+		// Read the diagnostic variables
+		auto pnh = ros::NodeHandle(ros::NodeHandle("~esc_telemetry"), "diagnostics");
+		lim =  new Limits(pnh);
```

Comment:
I think we shouldn't use pointer. Each ESC Diag object can have it's own copy.
Or we could also use const ref.

```
class ESCDiag {
  ESCDiag(std::string name, const Limit &lim_):
    DiagTask(name),
    lim(lim_) {
    }
    
const Limits &lim;
};
```

Reply:
And something like:

```
std::vector<ESCDiag> v;

v.emplace_back("name 1", lim);

Reply:
done, but it now no longer compiles, and the errors are .... hard to decode